pppd/plugins/pppol2tp/pppol2tp.c

```diff
@@ -241,10 +241,10 @@ static void send_config_pppol2tp(int mtu,
 		sprintf(&reorderto[0], "%d ", pppol2tp_reorder_timeout);
 	tid[0] = '\0';
 	if (pppol2tp_tunnel_id > 0)
-		sprintf(&tid[0], "%u ", pppol2tp_tunnel_id);
+		sprintf(&tid[0], "%hu ", (short) pppol2tp_tunnel_id);
```

Comment:
This is still incorrect for two reasons:
* `%hu` expects `unsigned short`, but you are passing `signed short` type cased from `signed int` type
* `pppol2tp_tunnel_id` is user parameter of (32-bit) integer type, so this change do some truncatenation

Anyway, this is used only for logging purposes and moreover L2TP tunnel id is 16-bit unsigned number, so it needs some more work to correctly fix these issues (add validation on input and e.g. disallow specifying too big number).

But in any case for debug / logging purposes, application should print what is really stored in variable, and not truncated content of variable. So I think that correct format should be really `%u` (or better `%d` as `pppol2tp_session_id` is `int`) and `pppol2tp_session_id` should be filled without any truncatenation.

Reply:
Looking at this code and warning is probably about small array size. So I think it would be better to really increase size of `tid` array instead of juggling with format types, casting and truncating to mute compile warning (which is fragile after 10 years nobody would really understand why value is truncated).